app/models/tasks/set_descriptors_handler.rb

```diff
@@ -72,7 +72,7 @@ def do_set_descriptors_task(_task, params) # rubocop:todo Metrics/CyclomaticComp
               # params[:fields] => <ActionController::Parameters {"1"=>"Operator", "2"=>"Workflow (Standard or Xp)", "3"=>"DPX1", "4"=>"DPX2", "5"=>"DPX3", "6"=>"NovaSeq Xp Mainfold", "7"=>"Pipette Carousel", "8"=>"PhiX lot #", "9"=>"PhiX %", "10"=>"Lane loading concentration (pM)", "11"=>"Comment"} permitted: true>
               # I believe that some of this complexity predates ordered hashes in ruby, and was an attempt to maintain field order.
               unless params[:descriptors].nil?
-                event.descriptors = params[:descriptors]
+                event.descriptors = params[:descriptors].to_unsafe_hash
```

Comment:
what's the reason for this?

Reply:
Descriptors are key value pairs defined by the task, and just get stored as a serialised hash. While we could potentially reconstruct the range of permitted keys from the task, there doesn't seem to be much value in doing so, as the only real attack here is adding new keys. I suppose potentially there is a risk of adding a potentially misleading novel key into the mix, 'witnessed by' or something, but it didn't seem worth the added complexity.